src/param/src/lib.rs

```diff
@@ -128,26 +136,84 @@ impl Param {
 
             // Parse value.
             match key {
-                b"TITLE" => title = Some(Self::read_utf8(&mut raw, i, format, len, 128)?),
-                b"TITLE_ID" => title_id = Some(Self::read_utf8(&mut raw, i, format, len, 12)?),
+                b"APP_VER" => {
+                    app_ver = Some(Self::read_utf8(&mut raw, i, format, len, 8)?);
+                }
+                b"CATEGORY" => {
+                    let category_param = Self::read_utf8(&mut raw, i, format, 4, 4)?;
+                    category = Some(category_param.clone());
+                    if !category_param.contains("bd") && !category_param.contains("ac")  // Blu-Ray Game, DLC.
+                        // Check if this is a patch
+                        && !category_param.starts_with("gp")
+                        // Check if this is an application
+                        && !category_param.starts_with("gd")
+                    {}
+                }
+                b"CONTENT_ID" => {
+                    content_id = Some(Self::read_utf8(&mut raw, i, format, len, 48)?);
+                }
+                b"TITLE" => {
+                    title = Some(Self::read_utf8(&mut raw, i, format, len, 128)?);
+                }
+                b"TITLE_ID" => {
+                    title_id = Some(Self::read_utf8(&mut raw, i, format, len, 12)?);
+                }
+                b"VERSION" => {
+                    version = Some(Self::read_utf8(&mut raw, i, format, len, 8)?);
+                }
                 _ => continue,
             }
         }
 
         Ok(Self {
-            title: title.ok_or(ReadError::MissingTitle)?,
-            title_id: title_id.ok_or(ReadError::MissingTitleId)?,
+            // App_Ver for Games and Patches, for DLC, use version. Anything else is abnormal.
+            app_ver: app_ver,
+            category: category.ok_or(ReadError::MissingCategory)?,
+            content_id: content_id.ok_or(ReadError::MissingContentId)?,
+            title: title,
+            title_id: title_id,
```

Comment:
So these fields can be absent? How do we know which game the patch belong to?

Reply:
APP_VER is available on Base App, Game Patch, and PS2 classics, but not on DLC as it doesn't change the version of the game. As for Title and Title_ID, they exist for apps, patches, DLC, and PS2 classics, but not any other category.

Reply:
> As for Title and Title_ID, they exist for apps, patches, DLC, and PS2 classics, but not any other category.

Does not mean it is always exists? Are there any other categories from this list?

Reply:
From what I saw, mainly savedata? Lemme double check.

Reply:
So, correction

```
ATTRIBUTE=0x00000000
CATEGORY=sd
DETAIL=
FORMAT=obs
MAINTITLE=Amplitude Saved Data
SAVEDATA_DIRECTORY=slot017
SAVEDATA_LIST_PARAM=0x00000000
SUBTITLE=
TITLE_ID=CUSA02480
```

It doesn't have App_Ver or Title, but the savefile does have Title_ID

Reply:
In this case better to make `TITLE_ID` as a mandatory.

Reply:
Already changing it back lol